_posts/2020-09-07-spring-boot-health-check.md

```diff
@@ -0,0 +1,365 @@
+
+**Micrometer is a library for collecting metrics from JVM-based applications and converting them in a format accepted by the monitoring tools.** It is a facade between application metrics and the metrics infrastructure developed by different monitoring systems like Prometheus, New Relic, and [many others](https://docs.spring.io/spring-boot/docs/current/reference/html/production-ready-features.html#production-ready-metrics). 
+
+To illustrate, we will integrate our Spring Boot application with one of these monitoring systems - Prometheus. **Prometheus operates on a pull model by scraping metrics from an endpoint exposed by the application instances at fixed intervals.**
+
+We will first add the micrometer SDK for Prometheus:
+```xml
+    <dependency>
+      <groupId>io.micrometer</groupId>
+      <artifactId>micrometer-registry-prometheus</artifactId>
+    </dependency>
+
+```
+We can integrate with another monitoring system like New Relic similarly by adding `micrometer-registry-newrelic` dependency for metric collection. New Relic in contrast to Prometheus works on a push model so we need to additionally configure credentials for New Relic in Spring Boot application.
+
+Continuing with our example with Prometheus, we will expose the Prometheus endpoint by updating the `management.endpoints.web.exposure.include` property in our application.properties.
+
+```application.properties
+management.endpoints.web.exposure.include=health,info,prometheus
+```
+Next, we will add the job in Prometheus with the configuration for scraping the metrics from our application.
+```yml
+  - job_name: 'user sign up'
+    metrics_path: '/actuator/prometheus'
+    scrape_interval: 5s
+    static_configs:
+    - targets: ['<HOST_NAME>:8080']
+```
+This configuration will scrape the metrics at 5-second intervals.
+
+We will use Docker to run Prometheus:
+```shell
+docker run \
+-p 9090:9090 \
+-v prometheus-config.yml:/etc/prometheus/prometheus.yml \
+prom/prometheus
+```
+Now we can check our application as a target in Prometheus:
+![Prometheus Targets](/assets/img/posts/spring-boot-health-check/prometheus-ss.png)
+
+As stated above, due to the Micrometer metrics facade we can integrate with other [monitoring tools](https://docs.spring.io/spring-boot/docs/current/reference/html/production-ready-features.html#production-ready-metrics) only by adding the provider-specific Micrometer dependency to the application. 
+
+## Visualizing Application Health
+It is easier to understand the metrics with the help of graphs and dashboards which enable us to see the trends over different time windows. We will visualize the metrics collected by our application using Grafana.
+
+![Grafana Dashboard](/assets/img/posts/spring-boot-health-check/grafana-ss.png)
+
+We are using a standard JVM dashboard to view our application metrics.
```

Comment:
This shows JVM metrics, not the health. 

I think we can remove this section, as it doesn't provide any value.